### PR TITLE
Finish up examples

### PR DESCRIPTION
**Pull request by jvanhoefer**
This PR finishes up the examples. Features:

- [x] Features a second realistic application examples (Finite State Projection)
- [x] Restructures the Lotka Volterra parts (one directory...)

### CODE REVIEW

File: doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Finite State Projection of a Gene Transmission Model\n",
+    "\n",
+    "## Model\n",
+    "\n",
+    "In this application example, we consider the two-stage model of gene transmission, dicussed e.g. in. [[1]](#References). The two model describes the stochastic transcription and translation of a gene by the reactions\n",
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb:11]:
```suggestion
    "In this application example, we consider the two-stage model of gene transmission, discussed e.g. in. [[1]](#References). The two-stage model describes the stochastic transcription and translation of a gene by the reactions\n",
```

File: doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    "\n",
+    "\n",
+    "\\begin{align*}\n",
+    "R_1 &: \\emptyset \\to mRNA \\\\\n",
+    "R_2 &: mRNA \\to \\emptyset\\\\\n",
+    "R_3 &: mRNA \\to mRNA + P \\\\\n",
+    "R_4 &: P \\to \\emptyset.\n",
+    "\\end{align*}\n",
+    "\n",
+    "The solution to the Chemical Master Equation (CME) [Ref TODO](#References) gives the probabilities of the states at a given time point as the solution of an infinite dimensional ODE\n",
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb:21]: TODO

File: doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    "\n",
+    "\\begin{align*}\n",
+    "\\frac{d}{dt}x_{r, p} &= - \\big(k_1 + (k_2+k_3) \\cdot r + k_4 \\cdot p \\big) \\cdot x_{r, p} \\\\\n",
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb:24]:
```suggestion
    "\\frac{d}{dt}x_{r, p} =& - \\big(k_1 + (k_2+k_3) \\cdot r + k_4 \\cdot p \\big) \\cdot x_{r, p} \\\\\n",
```

File: doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    "\n",
+    "We assume the initial probabilites to be independent Poisson distributions for mRNA and Protein abundances.\n",
+    "\n",
+    "## Finite State Projection\n",
+    "\n",
+    "The Finite State Projection approximates this CME by restricting the states to a finite domain and approximating the probability of the left out states by zero. In our case, we ill restrict the states to $x_{r, p}$ for $(r, p) \\in [0, r_{max}-1] \\times [[0, p_{max}-1]]$.\n",
+    "\n",
+    "## Model Construction:\n",
+    "\n",
+    "The model is constructed using `yaml2sbmls` Model Editor and exported to SBML. This takes only a few lines of code: "
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb:39]:
```suggestion
    "The model is constructed using the `yaml2sbml` Model Editor and exported to SBML. This takes only a few lines of code: "
```

File: doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    "[1] Shahrezaei, V.,  and Swain, P.S. (2008). \"Analytical distributions for stochastic gene expression\" PNAS 17256–17261 https://www.pnas.org/content/105/45/17256"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb:140]:
```suggestion
```

File: doc/examples/Finite_State_Projection/utils.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        r_marginal = [simulation.y[i, get_obs_idx_by_id(f'x_r{r}')] for r in range(r_max)]
+        p_marginal = [simulation.y[i, get_obs_idx_by_id(f'x_p{p}')] for p in range(p_max)]
+        # rna
+        plt.subplot(n_t, 2, 2*i+1)
+        plt.fill_between(np.arange(r_max), 0, r_marginal, facecolor='blue', alpha=0.5)
+        plt.plot(r_marginal)
+        plt.ylabel(f't={int(t[i])}')
+        plt.yticks([])
+        plt.xlim(0, r_max-1)
+        plt.ylim(0, 0.3)
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/utils.py:40]: Upper limit of the mRNA plot cuts off the peak

File: doc/examples/Finite_State_Projection/utils.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    simulation, model = compile_and_simulate(sbml_dir,
+                                             t,
+                                             r_max,
+                                             p_max)
+
+    def get_obs_idx_by_id(obs_id: str):
+        for idx, obs in enumerate(model.getObservableIds()):
+            if obs == obs_id:
+                return idx
+        raise IndexError(f'No observable with id {obs_id}')
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/utils.py:27]:
```suggestion
        try:
            return model.getObservableIds().index(obs_id)
        except ValueError:
            raise IndexError(f'No observable with id {obs_id}')
```

Same logic but may be faster/more conventional

File: doc/examples/README.md [hunk 8]
```diff
@@ -4,18 +4,20 @@ The [`yaml2sbml`](https://github.com/yaml2sbml-dev/yaml2sbml) package translates
...
 
-* [Lotka Volterra (Python) ](./Lotka_Volterra_python/Lotka_Volterra.ipynb)
-    * Introduces the input format, syntax & capabilities of `yaml2sbml`
-    * Showcases ODE simulation & parameter fitting using [AMICI](https://github.com/AMICI-dev/AMICI) & [pyPESTO](https://github.com/ICB-DCM/pyPESTO)
-* [Lotka Volterra CLI](./Lotka_Volterra_CLI/Lotka_Volterra_CLI.ipynb)
-    * Shows the command line interface of `yaml2sbml`
-* [Lotka Volterra Model Editor](.Lotka_Volterra_Model_Editor/Lotka_Volterra_Model_Editor.ipynb)
-    * Shows model construction using the model editor of `yaml2sbml`
-* [Sorensen](./Sorensen.yaml2sbml_Sorensen.ipynb)
+* [Lotka Volterra Notebooks]
```
Comment (dilpath) [inline doc/examples/README.md:7]:
```suggestion
* _Lotka Volterra Notebooks_
```
Maybe italicized looks nicer here

File: doc/examples/README.md [hunk 9]
```diff
@@ -4,18 +4,20 @@ The [`yaml2sbml`](https://github.com/yaml2sbml-dev/yaml2sbml) package translates
...
+    * [Lotka Volterra Model Editor](./Lotka_Volterra/Lotka_Volterra_Model_Editor/Lotka_Volterra_Model_Editor.ipynb)
+        * Shows model construction using the model editor of `yaml2sbml`
+* [Sorensen](./Sorensen/yaml2sbml_Sorensen.ipynb)
     * Application example, model of glucose and insulin metabolism
     * Shows the extension of an existing YAML model by the model editor
-* Pom1p (planned, work in progress)
-    * Application example, discretization of a PDE model of Pom1 gradient formation
+* [Finite State Projection](./Finite_State_Projection/Finite_State_Projection.ipynb)
+    * Application example of a stochastic gene transcription model having hundreds of states.
+    * Generates a complex and realistic model within a few lines of python.
```
Comment (dilpath) [inline doc/examples/README.md:20]:
```suggestion
    * Generates a complex and realistic model within a few lines of Python.
```

File: doc/examples/README.md [hunk 10]
```diff
@@ -4,18 +4,20 @@ The [`yaml2sbml`](https://github.com/yaml2sbml-dev/yaml2sbml) package translates
...
 * [Format Features](./Format_Features/Format_Features.ipynb)
     * several didactic examples, that show individual features of `yaml2sbml`:
```
Comment (dilpath) [inline doc/examples/README.md:22]:
```suggestion
    * Several didactic examples, that show individual features of `yaml2sbml`:
```

File: doc/examples/Sorensen/utils.py [hunk 11]
```diff
@@ -83,10 +83,10 @@ def compile_model(sbml_dir: str):
     }
 
     # Compile the model with AMICI.
-    sbml_importer = amici.sbml_import.SbmlImporter('Fig71_Sorensen1985.xml')
+    sbml_importer = amici.sbml_import.SbmlImporter(sbml_dir)
     sbml_importer.sbml2amici(
         model_name='Fig71_Sorensen1985',
-        output_dir=os.path.join('amici_models', 'Fig71_Sorensen1985'),
+        output_dir=os.path.join('amici_models', sbml_dir),
```
Comment (dilpath) [inline doc/examples/Sorensen/utils.py:89]: Output path would resolve to `amici_models/Fig71_Sorensen1985.xml/`, which is an odd name for a path, but probably still works (not sure if it `.` is OK in folder names in all popular operating systems though).

**Review by dilpath — APPROVED** [review]
:+1:

File: doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# References\n",
+    "\n",
+    "[1] Shahrezaei, V.,  and Swain, P.S. (2008). \"Analytical distributions for stochastic gene expression\" PNAS 17256–17261 https://www.pnas.org/content/105/45/17256"
+   ]
+  },
```
Comment (dilpath) [inline doc/examples/Finite_State_Projection/Finite_State_Projection.ipynb:133]:
```suggestion
  }
```

**Comment by jvanhoefer** [comment]
Thank you @dilpath

**Pull request merged**